src/daos/metadata_dao.py

```diff
@@ -0,0 +1,73 @@
+import csv
+import os
+from datetime import datetime
+from typing import Dict, Any, List
+import logging
+
+logger = logging.getLogger(__name__)
+
+FIELD_NAMES = ['market_slug', 'market_id', 'asset_id', 'outcome_name', 'executed_at_timestamp', 'game_start_timestamp']
+
+def _get_file_path(market_slug: str, executed_at: datetime) -> str:
+    """Get the file path for the metadata CSV file."""
+    csv_filename = os.path.join('data', f"{executed_at.strftime('%Y%m%d')}_{market_slug}_market-metadata.csv")
+    return csv_filename
+
+def _setup_csv(file_path: str) -> None:
+    """Set up the CSV file with headers if it doesn't exist."""
+    if not os.path.exists(file_path):
+        with open(file_path, 'w', newline='') as csvfile:
+            writer = csv.DictWriter(csvfile, fieldnames=FIELD_NAMES, quotechar='|', quoting=csv.QUOTE_MINIMAL)
+            writer.writeheader()
+        logger.info(f"Created new metadata CSV file: {file_path}")
+
+def _write_to_csv(file_path: str, rows: List[Dict[str, Any]]) -> None:
+    """Write rows to the CSV file."""
+    with open(file_path, 'a', newline='') as csvfile:
+        writer = csv.DictWriter(csvfile, fieldnames=FIELD_NAMES, quotechar='|', quoting=csv.QUOTE_MINIMAL)
+        writer.writerows(rows)
+
+def write_metadata(
+    market_slug: str,
+    market_id: int,
+    books: List[SyntheticOrderBook],
+    executed_at: datetime
+) -> None:
+    """
+    Write market metadata to CSV file.
+    
+    Args:
+        market_slug: The market slug identifier
+        market_id: The market ID
+        books: List of SyntheticOrderBook objects containing asset and outcome info
+        executed_at: Timestamp when SignalDrift executed
+    """
+    try:
+        file_path = _get_file_path(market_slug, executed_at)
+        _setup_csv(file_path)
+        
+        # Convert executed_at to epoch timestamp
+        executed_at_timestamp = int(executed_at.timestamp())
```

Comment:
```suggestion
        executed_at_timestamp = datetime_to_epoch(executed_at)
```
Use the `datetime_to_epoch` method we defined in /utils.